metpy/calc/indices.py

```diff
@@ -101,3 +101,132 @@ def mean_pressure_weighted(pressure, *args, **kwargs):
         ret.append(arg_mean * datavar.units)
 
     return ret
+
+
+@exporter.export
+@check_units('[pressure]', '[speed]', '[speed]', '[length]')
+def bunkers_storm_motion(pressure, u, v, heights):
+    r"""Calculate the Bunkers right-mover and left-mover storm motions and sfc-6km mean flow.
+
+    Uses the storm motion calculation from [Bunkers et al, 2000]_.
+
+    Parameters
+    ----------
+    pressure : array-like
+        Pressure from sounding
+    u : array-like
+        U component of the wind
+    v : array-like
+        V component of the wind
+    heights : array-like
+        Heights from sounding
+
+    Returns
+    -------
+    RM_vector
+        U and v component of Bunkers RM storm motion
+    LM_vector
+        U and v component of Bunkers LM storm motion
+    mean_vector
+        U and v component of sfc-6km mean flow
+
+    """
+    ums = u.to('m/s')
+    vms = v.to('m/s')
+
+    # mean wind from sfc-6km
+    u6m, v6m = mean_pressure_weighted(pressure, ums, vms, heights=heights,
+                                      depth=6000 * units('meter'))
+
+    # mean wind from sfc-500m
+    u5m, v5m = mean_pressure_weighted(pressure, ums, vms, heights=heights,
+                                      depth=500 * units('meter'))
+
+    # mean wind from 5.5-6km
+    u55m, v55m = mean_pressure_weighted(pressure, ums, vms, heights=heights,
+                                        depth=500 * units('meter'),
+                                        bottom=heights[0] + 5500 * units('meter'))
+
+    # Calculate the shear vector from sfc-500m to 5.5-6km
+    u6shr = u55m - u5m
+    v6shr = v55m - v5m
+
+    # making the shear vector
+    vl = [u6shr.magnitude, v6shr.magnitude, 0]
+
+    # Create a k hat vector
+    vk = [0, 0, 1]
+
+    # Take the cross product of the wind shear and k, and divide by the vector magnitude and
+    # multiply by the deviaton empirically calculated in Bunkers (2000) (7.5 m/s)
+    rdev = np.cross(vl, vk) * (7.5 / (np.sqrt(u6shr.magnitude ** 2 + v6shr.magnitude ** 2)))
```

Comment:
Isn't `np.cross(vl, vk)` in this case just the same has having:
```python
from metpy.units import concatenate
shear_cross = concatenate([shear[1], -shear[0]])
```
?

(That `concatenate` preserves units.)

Then you can do:
```python
rdev = shear_cross * (7.5 * units('m/s') / np.hypot(*shear))
```